spot_driver/spot_driver/spot_alerts.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+from tkinter import messagebox
+from typing import Any, List, Optional
+
+import bdai_ros2_wrappers.process as ros_process
+from bdai_ros2_wrappers.node import Node
+from rclpy.parameter import Parameter
+
+from spot_msgs.msg import (  # type: ignore
+    BatteryStateArray,
+)
+
+
+class SpotAlerts(Node):
+    """Monitors spot states and will generate a pop-up message to alert the user of warnings"""
+
+    def __init__(self, **kwargs: Any) -> None:
+        super().__init__("spot_alerts", **kwargs)
+        # Subscribers #
+        self.battery_states = self.create_subscription(
+            BatteryStateArray, "status/battery_states", self.battery_callback, 1
+        )
+
+        # Parameters #
+        self.declare_parameter("low_battery", False)
+
+    def battery_callback(self, battery_array_msg: BatteryStateArray) -> None:
+        """A callback function to check the battery percentage and send a pop-up alert when <= 10%
+        Args:
+            battery_array_msg: BatteryStateArray message received from the publisher containing BatteryState messages
+        Returns:
+            None
+        """
+        for battery_msg in battery_array_msg.battery_states:
+            if battery_msg.charge_percentage <= 10 and not self.get_parameter("low_battery").value:
```

Comment:
nit: might be nice to have 10 as a ros parameter vs hardcoded in case anyone wants to customize?

Reply:
ROS parameters are useful for setting user-configurable thresholds at startup, so I think this would be a good thing to expose as a parameter with 10 as the default value. You wouldn't necessarily need to modify the launchfile to allow the user to set it at runtime as part of this PR, either.

Reply:
Ok fair enough. I just couldnt really think of a scenario in which the user would need to change the low battery threshold while writing this lol

Reply:
I don't have an especially strong opinion about this either way. I think it's OK as you have it written right now.